lib/reqd.ml

```diff
@@ -192,6 +192,30 @@ let send_fixed_response t s response data =
   | Fixed _ | Complete _ ->
     failwith "h2.Reqd.respond_with_*: response already complete"
 
+let send_trailers_on_close t new_trailers =
+  let go s =
+    match s.response_state with
+    | Streaming (rsp, rsp_body, _old_trailers) ->
+      s.response_state <- Streaming (rsp, rsp_body, Some new_trailers)
+    | _ ->
+      failwith
+        "h2.Reqd.send_trailers_on_close: can only send trailers in Streaming \
+         mode"
+  in
+  match t.state with
+  | Idle | Active (Open (WaitingForPeer | PartialHeaders _), _) | Closed _ ->
+    assert false
```

Comment:
I think it's possible, in certain cases, to retain a reference to the request descriptor when the stream has been closed (e.g. because of an error). The other cases seem fine to me, but the `Closed` case should probably be handled separately.